cpp/src/arrow/python/numpy_convert.cc

```diff
@@ -412,6 +438,35 @@ Status NdarraysToSparseCSRMatrix(MemoryPool* pool, PyObject* data_ao, PyObject*
   return Status::OK();
 }
 
+Status NdarraysToSparseCSCMatrix(MemoryPool* pool, PyObject* data_ao, PyObject* indptr_ao,
```

Comment:
Basically this is the same as `NdarraysToSparseCSRMatrix`, but creating a `SparseCSCIndex` rather than a `SparseCSRIndex`? Perhaps make a common templatized helper?

Reply:
Done. Please check.